src/utils/db.ts

```diff
@@ -0,0 +1,48 @@
+import { Level } from "level";
+
+export type DBLevel = Level<string, string>;
+
+export default class DBService {
+  protected db: DBLevel;
+
+  private static _instance: DBService = new DBService();
+
+  protected constructor() {
+    if (DBService._instance) {
+      throw new Error(
+        "Error: Instantiation failed: Use DBService.getInstance() instead of new."
+      );
+    }
+    DBService._instance = this;
+
+    if (process.env.NODE_IS_TEST === "true") {
+      this.db = new Level<string, string>("./database_test", {
+        valueEncoding: "json",
+        createIfMissing: true,
+        errorIfExists: false,
+      });
+    } else {
+      this.db = new Level<string, string>("./database", {
+        valueEncoding: "json",
+        createIfMissing: true,
+        errorIfExists: false,
+      });
```

Comment:
Nitpick: options are identical. Create a const for it an use it on both

Reply:
Refactored to now receive database location from the constructor.